### PR TITLE
Add modification to replace tie lines by lines

### PR DESCRIPTION
**Pull request by flo-dup**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

No

**What kind of change does this PR introduce?**

Feature

**What is the new behavior (if this is a feature change)?**
A tie line can be replaced by a line thanks to a network modification

### CODE REVIEW

File: iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/TieLineUtil.java [hunk 1]
```diff
@@ -77,28 +84,37 @@ public static void mergeProperties(DanglingLine dl1, DanglingLine dl2, Propertie
                 properties.setProperty(prop, dl1.getProperty(prop));
             } else if (dl1.getProperty(prop).isEmpty()) {
                 LOGGER.debug("Inconsistencies of property '{}' between both sides of merged line. Side 1 is empty, keeping side 2 value '{}'", prop, dl2.getProperty(prop));
+                propertyOnlyOnOneSide(reporter, prop, dl2.getProperty(prop), 1, dl1.getId(), dl2.getId());
                 properties.setProperty(prop, dl2.getProperty(prop));
             } else if (dl2.getProperty(prop).isEmpty()) {
                 LOGGER.debug("Inconsistencies of property '{}' between both sides of merged line. Side 2 is empty, keeping side 1 value '{}'", prop, dl1.getProperty(prop));
+                propertyOnlyOnOneSide(reporter, prop, dl1.getProperty(prop), 2, dl1.getId(), dl2.getId());
```
Comment (flo-dup) [inline iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/TieLineUtil.java:91]: Shouldn't it be:
```suggestion
                propertyOnlyOnOneSide(reporter, prop, dl1.getProperty(prop), 2, dl2.getId(), dl1.getId());
```

Comment (miovd) [reply iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/TieLineUtil.java:91]: No it should be always side 1 first then side 2

**Comment by miovd** [comment]
@flo-dup you deleted MergedXnode and it doesn't work anymore :( Should we revert it?

**Comment by miovd** [comment]
> @flo-dup you deleted MergedXnode and it doesn't work anymore :( Should we revert it?

It had been decided to keep it deleted and only add aliases

**Comment by annetill** [comment]
We should remain coherent with the previous network modifications. In `ReplaceTeePointByVoltageLevelOnLine` and in `ConnectVoltageLevelOnLine`, we remove lines without making any treatment on extension, aliases or properties. It was a question of course at that moment, for example for short-circuit extensions, positions extensions, etc. As we decided not to do it at a first approach, I think that you have to get ride of everything at a first approach. I just have a doubt with dangling lines id that could be added as aliases.

**Comment by miovd** [comment]
> We should remain coherent with the previous network modifications. In `ReplaceTeePointByVoltageLevelOnLine` and in `ConnectVoltageLevelOnLine`, we remove lines without making any treatment on extension, aliases or properties. It was a question of course at that moment, for example for short-circuit extensions, positions extensions, etc. As we decided not to do it at a first approach, I think that you have to get ride of everything at a first approach. I just have a doubt with dangling lines id that could be added as aliases and maybe the Xnode name.

Contrary to `ReplaceTeePointByVoltageLevelOnLine` or `ConnectVoltageLevelOnLine`, this modification is foremost dedicated to keep backwards compatibility, which is why I made it reproduce former behavior with aliases and properties merging. Extensions are not kept, only a warning is ensued.

File: iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/ReplaceTieLinesByLinesPostProcessor.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import com.powsybl.iidm.network.ImportPostProcessor;
+import com.powsybl.iidm.network.Network;
+
+/**
+ * @author Miora Vedelago <miora.ralambotiana at rte-france.com>
+ */
+@AutoService(ImportPostProcessor.class)
+public class ReplaceTieLinesByLinesPostProcessor implements ImportPostProcessor {
+
+    public static final String THROW_EXCEPTION = "iidm.import.post-processor.replace-tie-lines-by-lines.throw-exception";
```
Comment (annetill) [inline iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/ReplaceTieLinesByLinesPostProcessor.java:26]: @annetill @flo-dup @olperr1  to be documented in the migration guide.

**Review by annetill — APPROVED** [review]

File: iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/ReplaceTieLinesByLines.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+            double p1 = dl1.getTerminal().getP();
+            double q1 = dl1.getTerminal().getQ();
+            double p2 = dl2.getTerminal().getP();
+            double q2 = dl2.getTerminal().getQ();
+            Properties properties = new Properties();
+            tl.getPropertyNames().forEach(pn -> properties.put(pn, tl.getProperty(pn)));
+            mergeProperties(dl1, dl2, properties, reporter);
+            Map<String, String> aliases = new HashMap<>();
+            tl.getAliases().forEach(alias -> aliases.put(alias, tl.getAliasType(alias).orElse("")));
+            mergeIdenticalAliases(dl1, dl2, aliases, reporter);
```
Comment (flo-dup) [inline iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/ReplaceTieLinesByLines.java:84]: There cannot be identical aliases, dl1 and dl2 are in the same network. An exception is thrown when an existing alias is added in the network index.

**Pull request merged**